File: frontend/demo/component/passwordfield/password-field-basic-features.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/password-field';
import '@vaadin/icon';
import '@vaadin/icons';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('password-field-basic-features')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
        <!-- tag::snippet[] -->
        <vaadin-password-field
          label="Label"
          helper-text="Helper text"
          placeholder="Placeholder"
          clear-button-visible>
          <vaadin-tooltip slot="tooltip" text="Tooltip text"></vaadin-tooltip>
          <vaadin-icon slot="prefix" icon="vaadin:lock"></vaadin-icon>
        </vaadin-password-field>
        <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/passwordfield/PasswordFieldBasicFeatures.java
package com.vaadin.demo.component.passwordfield;

import com.vaadin.flow.component.icon.VaadinIcon;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.textfield.PasswordField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("password-field-basic-features")
public class PasswordFieldBasicFeatures extends HorizontalLayout {

    public PasswordFieldBasicFeatures() {
        setPadding(false);

        // tag::snippet[]
        PasswordField field = new PasswordField();
        field.setLabel("Label");
        field.setHelperText("Helper text");
        field.setPlaceholder("Placeholder");
        field.setTooltipText("Tooltip text");
        field.setClearButtonVisible(true);
        field.setPrefixComponent(VaadinIcon.LOCK.create());
        // end::snippet[]

        add(field);
    }

    public static class Exporter extends DemoExporter<PasswordFieldBasicFeatures> { // hidden-source-line
    } // hidden-source-line
}
